### PR TITLE
integration: test for file discovery

### PR DESCRIPTION
**Pull request by chanwit**
This is an integration test for file discovery. This PR fixes issue #498.

### CODE REVIEW

File: test/integration/discovery.bats [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bats
+
+load helpers
+
+DISCOVERY_FILE=/tmp/test_file_discovery
```
Comment (vieux) [inline test/integration/discovery.bats]: can we use `mkfile` ?

File: test/integration/discovery.bats [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+function teardown() {
+	stop_manager
+	stop_docker
+	rm -f $DISCOVERY_FILE
+}
+
+function start_manager_with_file_discovery() {
+	rm -f $DISCOVERY_FILE
```
Comment (vieux) [inline test/integration/discovery.bats]: is this needed here ?

File: test/integration/discovery.bats (hunk 1 above)
Comment (chanwit) [inline test/integration/discovery.bats]: `mkfile` is not available everywhere. Maybe I'll use `cat /dev/null > /tmp/test_file_discovery` to create an empty file.

File: test/integration/discovery.bats (hunk 2 above)
Comment (chanwit) [inline test/integration/discovery.bats]: I'll make it better.

**Comment by chanwit** [comment]
@vieux PTAL, thank you.

**Comment by chanwit** [comment]
@vieux please hold on.
I should rewrite this after #582 getting merged.

**Comment by vieux** [comment]
@chanwit OK

File: test/integration/discovery.bats (hunk 1 above)
Comment (aluzzardi) [inline test/integration/discovery.bats]: Could we use `mktemp` instead?

The goal of the integration tests is to run them on Jenkins or something like that automatically after each merge, so we kinda control the environment.

**Comment by aluzzardi** [comment]
Agreed @chanwit, this will be heavily inspired from #582 once we are done with it - as every integration tests for every discovery service.

Very exciting!

File: test/integration/discovery.bats (hunk 1 above)
Comment (chanwit) [inline test/integration/discovery.bats]: I'll try `mktemp` this time.

File: test/integration/file-discovery.bats [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+function teardown() {
+	swarm_manage_cleanup
+	rm $DISCOVERY_FILE
+	stop_docker
+}
+
+function setup_file_discovery() {
+	# create a temp file
+	DISCOVERY_FILE=$(mktemp)
```
Comment (aluzzardi) [inline test/integration/file-discovery.bats]: **Maybe** it's better to put this outside of the function.

If the test fails before calling this (`start_docker` fails), `DISCOVERY_FILE` will be undefined and the teardown will `rm` with no arguments.

Not a huge deal.

File: test/integration/file-discovery.bats [hunk 4]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bats
+
+load helpers
+
+function teardown() {
+	swarm_manage_cleanup
+	rm $DISCOVERY_FILE
```
Comment (aluzzardi) [inline test/integration/file-discovery.bats]: It's probably worth using `rm -f`

File: test/integration/file-discovery.bats [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+function setup_file_discovery() {
+	# create a temp file
+	DISCOVERY_FILE=$(mktemp)
+	for host in ${HOSTS[@]}; do
+		echo "$host" >> $DISCOVERY_FILE
+	done
+}
+
+@test "docker info should return the number of nodes with file discovery" {
+	start_docker 3
```
Comment (aluzzardi) [inline test/integration/file-discovery.bats]: 2 nodes are probably enough for this

**Comment by aluzzardi** [comment]
Minor comments, otherwise LGTM

Thank you @chanwit !

**Pull request merged**